aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -185,6 +189,23 @@ module aptos_framework::code {
             request_publish(addr, module_names, code, policy.policy)
     }
 
+    public fun is_package_upgradeable(metadata_serialized: vector<u8>): bool {
+        let metadata = util::from_bytes<PackageMetadata>(metadata_serialized);
+        metadata.upgrade_policy.policy < upgrade_policy_immutable().policy
+    }
+
+    public fun freeze_package_registry(publisher: &signer, registry: Object<PackageRegistry>) acquires PackageRegistry {
```

Comment:
Can it be unfrozen? If yes, is there a function for that missing?

Either way, could you mention this in a comment above this?

Reply:
Will add more comments. To answer your question, it cannot be unfrozen as this function would not serve much purpose if it can be frozen and unfrozen any time.